pkg/driver/executor/executor.go

```diff
@@ -492,22 +495,10 @@ func (ex *Executor) getMachineByID(_ context.Context, serverID string) (*servers
 // to store tags in a respective field and do a server-side filtering. To avoid incompatibility with older versions
 // we will continue making the filtering clientside.
 func (ex *Executor) getMachineByName(_ context.Context, machineName string) (*servers.Server, error) {
-	var (
-		searchClusterName string
-		searchNodeRole    string
-	)
-
-	for key := range ex.Config.Spec.Tags {
-		if strings.Contains(key, cloudprovider.ServerTagClusterPrefix) {
-			searchClusterName = key
-		} else if strings.Contains(key, cloudprovider.ServerTagRolePrefix) {
-			searchNodeRole = key
-		}
-	}
-
-	if searchClusterName == "" || searchNodeRole == "" {
-		klog.Warningf("getMachineByName operation can not proceed: cluster/role tags are missing for machine [Name=%q]", machineName)
-		return nil, fmt.Errorf("getMachineByName operation can not proceed: cluster/role tags are missing for machine [Name=%q]", machineName)
+	searchClusterName, searchNodeRole, ok := findMandatoryTags(ex.Config.Spec.Tags)
+	if !ok {
+		klog.Warningf("operation can not proceed: cluster/role tags are missing")
+		return nil, fmt.Errorf("operation can not proceed: cluster/role tags are missing")
```

Comment:
we can enhance the error message to including `machineName` in the error message like it was before

Reply:
and also in the warning message